### PR TITLE
[jansson] Update to 2.14

### PR DESCRIPTION
**Pull request by Jimmy-Hu**
[jansson] Update to 2.14

**Describe the pull request**

- #### What does your PR fix? Update jansson package to 2.14
  
  https://github.com/akheron/jansson/releases/tag/v2.14

### CODE REVIEW

**Comment by Jimmy-Hu** [comment]
Not sure why it failing (`Version `2.14#2` was not found in versions file.`) in x86_windows

on my machine

```
vcpkg.exe install jansson
Computing installation plan...
The following packages will be built and installed:
    jansson[core]:x86-windows -> 2.14#2
Detecting compiler hash for triplet x86-windows...
Restored 0 packages from C:\Users\jimmyhu\AppData\Local\vcpkg\archives in 587.5 us. Use --debug to see more details.
Starting package 1/1: jansson:x86-windows
Building package jansson[core]:x86-windows...
-- Downloading https://github.com/akheron/jansson/archive/684e18c927e89615c2d501737e90018f4930d6c5.tar.gz -> akheron-jansson-684e18c927e89615c2d501737e90018f4930d6c5.tar.gz...
-- Extracting source V:/downloads/akheron-jansson-684e18c927e89615c2d501737e90018f4930d6c5.tar.gz
-- Using source at V:/buildtrees/jansson/src/8f4930d6c5-d0383a6ce3.clean
-- Found external ninja('1.10.2').
-- Configuring x86-windows
-- Building x86-windows-dbg
-- Building x86-windows-rel
-- Installing: V:/packages/jansson_x86-windows/share/jansson/copyright
-- Fixing pkgconfig file: V:/packages/jansson_x86-windows/lib/pkgconfig/jansson.pc
-- Using cached msys-mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst.
-- Using msys root at V:/downloads/tools/msys2/9a1ec3f33446b195
-- Fixing pkgconfig file: V:/packages/jansson_x86-windows/debug/lib/pkgconfig/jansson.pc
-- Performing post-build validation
-- Performing post-build validation done
Stored binary cache: C:\Users\jimmyhu\AppData\Local\vcpkg\archives\c9\c99fbe967b6b8fc71353603f8ab9a4971e557a8e09a3fb294d32eb96588adbc6.zip
Installing package jansson[core]:x86-windows...
Elapsed time for package jansson:x86-windows: 39.96 s

Total elapsed time: 42.56 s

The package jansson provides CMake targets:

    find_package(jansson CONFIG REQUIRED)
    target_link_libraries(main PRIVATE jansson::jansson)


```

**Review by JackBoosY — APPROVED** [review]

File: ports/jansson/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "jansson",
-  "version-string": "2.13.1",
+  "version-string": "2.14",
```
Comment (BillyONeal) [inline ports/jansson/vcpkg.json:3]:
```suggestion
  "version": "2.14",
```
?

File: versions/j-/jansson.json [hunk 2]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "4f2760bb6ef390f3472b4789914024054fccaeb6",
```
Comment (JackBoosY) [inline versions/j-/jansson.json:4]:
```suggestion
      "git-tree": "13cb17992f4a9ef3eaba47a1449bb859e9009a03",
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: versions/j-/jansson.json [hunk 3]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "13cb17992f4a9ef3eaba47a1449bb859e9009a03",
+      "version-string": "2.14",
```
Comment (JackBoosY) [inline versions/j-/jansson.json:5]:
```suggestion
      "version": "2.14",
```

**Review by JackBoosY — APPROVED** [review]

File: ports/jansson/vcpkg.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "jansson",
-  "version-string": "2.13.1",
+  "version": "2.14",
   "port-version": 2,
   "description": "Jansson is a C library for encoding, decoding and manipulating JSON data",
```
Comment (Thomas1664) [inline ports/jansson/vcpkg.json:5]:
```suggestion
  "version": "2.14",
  "description": "Jansson is a C library for encoding, decoding and manipulating JSON data",
```

**Review by Thomas1664 — CHANGES_REQUESTED** [review]
Please remove `port-version`

File: versions/j-/jansson.json [hunk 5]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "13cb17992f4a9ef3eaba47a1449bb859e9009a03",
+      "version": "2.14",
+      "port-version": 2
```
Comment (JackBoosY) [inline versions/j-/jansson.json:6]:
```suggestion
      "port-version": 0
```

Comment (JackBoosY) [inline versions/j-/jansson.json:6]:
```suggestion
      "git-tree": "1a5c55feab368e4bfb40be993b9ea5a39fc82f51",
      "version": "2.14",
      "port-version": 0
```

File: versions/baseline.json [hunk 6]
```diff
@@ -3009,7 +3009,7 @@
       "port-version": 1
     },
     "jansson": {
-      "baseline": "2.13.1",
+      "baseline": "2.14",
       "port-version": 2
```
Comment (JackBoosY) [inline versions/baseline.json:3013]:
```suggestion
      "port-version": 0
```

**Review by JackBoosY — APPROVED** [review]

**Review by raghavthind2005 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**